packages/core/src/emitter.ts

```diff
@@ -0,0 +1,311 @@
+	guildScheduledEventDelete: [WithIntrinsicProps<{ event: APIGuildScheduledEvent }>];
+	guildScheduledEventUpdate: [WithIntrinsicProps<{ event: APIGuildScheduledEvent }>];
+	guildScheduledEventUserAdd: [WithIntrinsicProps<{ data: GatewayGuildScheduledEventUserAddDispatchData }>];
+	guildScheduledEventUserRemove: [WithIntrinsicProps<{ data: GatewayGuildScheduledEventUserRemoveDispatchData }>];
+	guildStickersUpdate: [WithIntrinsicProps<{ data: GatewayGuildStickersUpdateDispatchData }>];
+	guildUpdate: [WithIntrinsicProps<{ guild: APIGuild }>];
+	integrationCreate: [WithIntrinsicProps<{ integration: GatewayIntegrationCreateDispatchData }>];
+	integrationDelete: [WithIntrinsicProps<{ data: GatewayIntegrationDeleteDispatchData }>];
+	integrationUpdate: [WithIntrinsicProps<{ integration: GatewayIntegrationUpdateDispatchData }>];
+	interactionCreate: [WithIntrinsicProps<{ interaction: APIInteraction }>];
+	inviteCreate: [WithIntrinsicProps<{ invite: GatewayInviteCreateDispatchData }>];
+	inviteDelete: [WithIntrinsicProps<{ data: GatewayInviteDeleteDispatchData }>];
+	messageCreate: [WithIntrinsicProps<{ message: APIMessage }>];
+	messageDelete: [WithIntrinsicProps<{ data: GatewayMessageDeleteDispatchData }>];
+	messageDeleteBulk: [WithIntrinsicProps<{ data: GatewayMessageDeleteBulkDispatchData }>];
+	messageReactionAdd: [WithIntrinsicProps<{ data: GatewayMessageReactionAddDispatchData }>];
+	messageReactionRemove: [WithIntrinsicProps<{ data: GatewayMessageReactionRemoveDispatchData }>];
+	messageReactionRemoveAll: [WithIntrinsicProps<{ data: GatewayMessageReactionRemoveAllDispatchData }>];
+	messageReactionRemoveEmoji: [WithIntrinsicProps<{ data: GatewayMessageReactionRemoveEmojiDispatchData }>];
+	messageUpdate: [WithIntrinsicProps<{ message: GatewayMessageUpdateDispatchData }>];
+	presenceUpdate: [WithIntrinsicProps<{ data: GatewayPresenceUpdate }>];
+	ready: [WithIntrinsicProps<{ data: GatewayReadyDispatchData }>];
+	stageInstanceCreate: [WithIntrinsicProps<{ stage_instance: APIStageInstance }>];
+	stageInstanceDelete: [WithIntrinsicProps<{ stage_instance: APIStageInstance }>];
+	stageInstanceUpdate: [WithIntrinsicProps<{ stage_instance: APIStageInstance }>];
+	threadCreate: [WithIntrinsicProps<{ thread: APIThreadChannel }>];
+	threadDelete: [WithIntrinsicProps<{ thread: APIThreadChannel }>];
+	threadListSync: [WithIntrinsicProps<{ data: GatewayThreadListSync }>];
+	threadMemberUpdate: [WithIntrinsicProps<{ data: GatewayThreadMemberUpdateDispatchData }>];
+	threadMembersUpdate: [WithIntrinsicProps<{ data: GatewayThreadMembersUpdateDispatchData }>];
+	threadUpdate: [WithIntrinsicProps<{ thread: APIThreadChannel }>];
+	userUpdate: [WithIntrinsicProps<{ user: APIUser }>];
+	voiceServerUpdate: [WithIntrinsicProps<{ data: GatewayVoiceServerUpdateDispatchData }>];
+	voiceStateUpdate: [WithIntrinsicProps<{ state: GatewayVoiceState }>];
+	webhooksUpdate: [WithIntrinsicProps<{ data: GatewayWebhooksUpdateDispatchData }>];
```

Comment:
Again, I don't know what the rest of the @discordjs/core team thinks, but I am against this inconsistency in how the data is presented. Most events pass `payload.d` in the `data` property, but some do it in a differently named property... I'd prefer we were consistent - that is either all named or all under `data`. *Personally* I'd prefer we kept everything under `data` as that makes it the cleanest.